### PR TITLE
[Method References 2] LPS-176936 Convert method reference to field reference in class LiferayResourceOwnerAccessTokenGrantHandler

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(3/1 9:31)

**Review by dantewang — APPROVED** [review]

**Comment by W2jx** [comment]
no conflicts(3/3 10:55)

**Comment by W2jx** [comment]
no conflicts(3/6 16:03)

**Comment by W2jx** [comment]
no conflicts(3/8 15:25)

**Comment by W2jx** [comment]
no conflicts(3/14 15:20)

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java [hunk 1]
```diff
@@ -104,40 +104,22 @@ protected boolean isGrantHandlerEnabled() {
...
+	private static final Log _log = LogFactoryUtil.getLog(
+		LiferayResourceOwnerAccessTokenGrantHandler.class);
+
 	@Reference(
 		policy = ReferencePolicy.DYNAMIC,
 		policyOption = ReferencePolicyOption.GREEDY
 	)
-	protected void setLiferayOAuthDataProvider(
-		LiferayOAuthDataProvider liferayOAuthDataProvider) {
+	private volatile LiferayOAuthDataProvider _liferayOAuthDataProvider;
```
Comment (tinatian) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java:114]: @W2jx This does not need to be dynaimc and greedy, as LiferayOAuthDataProvider is an internal class.

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java [hunk 2]
```diff
@@ -52,15 +52,15 @@
...
-		_resourceOwnerGrantHandler.setLoginHandler(_resourceOwnerLoginHandler);
-
 		_oAuth2ProviderConfiguration = ConfigurableUtil.createConfigurable(
 			OAuth2ProviderConfiguration.class, properties);
 	}
 
 	@Override
 	protected AccessTokenGrantHandler getAccessTokenGrantHandler() {
+		_resourceOwnerGrantHandler.setDataProvider(_liferayOAuthDataProvider);
+		_resourceOwnerGrantHandler.setLoginHandler(_resourceOwnerLoginHandler);
```
Comment (tinatian) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java:62]: @W2jx It is really weird that you have a local variable, and each time you retrieve it, you set it. Usually for those you initialize once and reuse later, we make it a local variable, here it should not be a local variable.

Also it is really weird that `LiferayAuthorizationAccessTokenCodeGrantHandler` is a `AccessTokenGrantHandler`, and it has a `getAccessTokenGrantHandler` method which did not return itself, but another type of `AccessTokenGrantHandler`.

You should not have `getAccessTokenGrantHandler()`, instead, let `LiferayAuthorizationAccessTokenCodeGrantHandler` directly use `AuthorizationCodeGrantHandler` to implement the logic.

FYI @dantewang

**Comment by tinatian** [comment]
@W2jx Please resend.

Thanks.
Tina

**Pull request closed without merging**